diff --git a/distrho/src/DistrhoPluginLV2export.cpp b/distrho/src/DistrhoPluginLV2export.cpp
index 4905b764..79ecc25e 100644
--- a/distrho/src/DistrhoPluginLV2export.cpp
+++ b/distrho/src/DistrhoPluginLV2export.cpp
@@ -1178,8 +1178,8 @@ void lv2_generate_ttl(const char* const basename)
             const uint32_t microVersion = (version & 0x0000FF) >> 0;
 
             // NOTE: LV2 ignores 'major' version and says 0 for minor is pre-release/unstable.
-            if (majorVersion > 0)
-                minorVersion += 2;
+            if (majorVersion > 0 && minorVersion == 0)
+                minorVersion = 2;
 
             pluginString += "    lv2:microVersion " + String(microVersion) + " ;\n";
             pluginString += "    lv2:minorVersion " + String(minorVersion) + " .\n";
